File: Practice/Greedy Alogirthm/FinishMaximumJobs.js
function FinishMaximumJobs(A, B) {
  let output = 1;
  let newArray = [];
  for (let i = 0; i < A.length; i++) {
    let curr_input = [A[i], B[i]];
    newArray.push(curr_input);
  }
  newArray.sort(function (a, b) {
    return a[1] - b[1];
  });
  let endTime = newArray[0][1];
  for (let i = 1; i < newArray.length; i++) {
    if (newArray[i][0] >= endTime) {
      output++;
      endTime = newArray[i][1];
    }
  }

  console.log(output);
}

A = [4, 4, 8, 15, 6];
B = [9, 5, 15, 16, 7];

FinishMaximumJobs(A, B);

File: Revision/Greedy/FinishMaximumJobs.java
import java.util.*;
public class FinishMaximumJobs {
    public int solve(ArrayList<Integer> A, ArrayList<Integer> B) {
        ArrayList<ArrayList<Integer>> mergeArrays = new ArrayList<>();

        for(int i=0;i<A.size();i++){
            ArrayList<Integer> input = new ArrayList<>();
            input.add(A.get(i));
            input.add(B.get(i));
            mergeArrays.add(input);
        }

        Collections.sort(mergeArrays, new customComparator());

        int count = 1;
        int endTime = mergeArrays.get(0).get(1);
        for(int i=1;i<mergeArrays.size();i++){
            if(mergeArrays.get(i).get(0) >= endTime){
                count++;
                endTime = mergeArrays.get(i).get(1);
            } 
            
        }

        return count;
    }

}

class customComparator implements Comparator<ArrayList<Integer>>{
    public int compare(ArrayList<Integer> A, ArrayList<Integer> B){
        return A.get(1) - B.get(1);
    }
}
